### PR TITLE
add filter for unreachable services

### PR DESCRIPTION
**Pull request by jabdoa2**
- Add `(UNREACHABLE SERVICE)` to the status line of unreachable services
- Add a new filter checkbox to filter unreachable services
- Filter out unreachable services when checkbox is checked
- Query `is_reachable` field for icinga2web

### CODE REVIEW

**Comment by jabdoa2** [comment]
@HenriWahl any chance to get this merged?

**Comment by HenriWahl** [comment]
Sorry for the late reply - I am about to check this.

File: Nagstamon/Objects.py [hunk 1]
```diff
@@ -147,6 +147,7 @@ class GenericService(GenericObject):
 
     def __init__(self):
         GenericObject.__init__(self)
+        self.unreachable = False
```
Comment (HenriWahl) [inline Nagstamon/Objects.py:150]: Sorry for the late reply.

Why has self.unreachable to be initialized here? Shouldn't it be set like other properties?

File: Nagstamon/QUI/settings_main.py [hunk 2]
```diff
@@ -1642,3 +1646,13 @@ def retranslateUi(self, settings_main):
         self.label_acknowledge_submit_result.setTitle(_translate("settings_main", "Submit check result:"))
         self.label_defaults_submit_check_result_comment.setText(_translate("settings_main", "Comment:"))
         self.tabs.setTabText(self.tabs.indexOf(self.tab_defaults), _translate("settings_main", "Defaults"))
+
+
+if __name__ == "__main__":
```
Comment (HenriWahl) [inline Nagstamon/QUI/settings_main.py:1651]: Where does this condition come from? Is it inserted by the GUI compiler pyuic5?

**Comment by HenriWahl** [comment]
Stupid question: what are unreachable services? As far as I understand "unreachable" applies only to hosts?

File: Nagstamon/QUI/settings_main.py (hunk 2 above)
Comment (jabdoa2) [reply Nagstamon/QUI/settings_main.py:1651]: Exactly

File: Nagstamon/Objects.py (hunk 1 above)
Comment (jabdoa2) [reply Nagstamon/Objects.py:150]: Because it applies to services only. We can move it up if you want

**Comment by jabdoa2** [comment]
> Stupid question: what are unreachable services? As far as I understand "unreachable" applies only to hosts?

Those are services with a dependency. For instance if a service is actually checked via Prometheus and that host is down it would fail. For those services we define a dependency so we can filter them out as unreachable.

**Pull request merged**